core/src/main/kotlin/riven/core/service/workflow/coordinator/DagExecutionCoordinator.kt

```
@@ -0,0 +1,279 @@
+            }
+
+            // Transition state: nodes ready for execution
+            state = StateTransition.apply(
+                state,
+                NodesReady(readyNodes.map { it.id }.toSet())
+            )
+
+            // Execute ready nodes in parallel (caller handles Temporal Async/Promise)
+            val results: List<Pair<UUID, Any?>> = try {
+                nodeExecutor(readyNodes)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)